### PR TITLE
autocopr: distro conditionals for containers-common

### PR DESCRIPTION
**Pull request by lsm5**
Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

@containers/podman-maintainers @Luap99 @mheon @rhatdan @baude @vrothberg @edsantiago @cevich PTAL.

This will get build automatically on merge.

#### What this PR does / why we need it:

Handle containers-common dependency in `rhcontainerbot/podman-next` copr.

#### How to verify it
```bash
$ sudo dnf copr enable rhcontainerbot/podman-next -y
$ sudo dnf install netavark podman -y
```
#### Which issue(s) this PR fixes:
None

#### Special notes for your reviewer:

None

### CODE REVIEW

**Review by Luap99 — COMMENTED** [review]
LGTM

File: .autocopr/podman.spec [hunk 1]
```diff
@@ -51,7 +51,11 @@ BuildRequires: ostree-devel
 BuildRequires: systemd
 BuildRequires: systemd-devel
 Requires: conmon >= 2:2.0.30-2
-Requires: containers-common >= 4:1-30
+%if 0%{?fedora}
+Requires: containers-common >= 4:1-21
```
Comment (edsantiago) [inline .autocopr/podman.spec:55]: Why the downgrade?

Comment (lsm5) [reply .autocopr/podman.spec:55]: @edsantiago f34 is on `4:1-21`, I figure a simple change should be ok since it's not affecting mainstream podman users.

Comment (lsm5) [reply .autocopr/podman.spec:55]: but if you prefer separation between f34 , f35 etc. I could do that.

Comment (Luap99) [reply .autocopr/podman.spec:55]: see podman IRC, f34 only has 1.21

Comment (edsantiago) [reply .autocopr/podman.spec:55]: Ack, thanks. This is fine, I just didn't have the context.

Comment (lsm5) [reply .autocopr/podman.spec:57]: ack, i noticed your comment just now, but made the finer changes already. PTAL.

**Comment by lsm5** [comment]
err, may need another fix.

File: .autocopr/podman.spec (hunk 1 above)
Comment (lsm5) [reply .autocopr/podman.spec:55]: @edsantiago na, I reverted to the earlier simpler change :) . PTAL now.

**Comment by cevich** [comment]
> This will get build automatically on merge.

Will it properly handle/differentiate across multiple podman branches, e.g. when `v4.0` branch is created with a copy of this file, will copr confuse it with changes on `main`?

**Comment by cevich** [comment]
Oh also, we have an existing `Test build RPM` CI task, should that be updated to use this spec file instead?

**Comment by lsm5** [comment]
@cevich don't know if you can see this page: https://copr.fedorainfracloud.org/coprs/rhcontainerbot/podman-next/package/podman/ but it has a field `Committish: main` which records the upstream branch to build from. When we want to build from v4.0, we just gotta switch that value over there. /cc @mheon @baude.

@cevich if you need admin access to the copr repo, add yourself to https://copr.fedorainfracloud.org/coprs/rhcontainerbot/podman-next/permissions/ and I can approve you .

@cevich RE: `Test build RPM` I'm not too sure, i think the existing task uses some magic in `hack/` + makefile targets or something, yes? Not sure it's easily transferrable to `.autocopr/podman.spec`

**Comment by cevich** [comment]
> field Committish: main which records the upstream branch to build from

Perfect.  I think maybe it doesn't need to change, as this package simply always represent the latest/greatest from 'main', no? 
 Otherwise I might suggest adding a step in `RELEASE_PROCESS.md` to update the branch field.

**Comment by lsm5** [comment]
> > field Committish: main which records the upstream branch to build from
> 
> Perfect. I think maybe it doesn't need to change, as this package simply always represent the latest/greatest from 'main', no? Otherwise I might suggest adding a step in `RELEASE_PROCESS.md` to update the branch field.

@mheon @baude wdyt?

**Comment by cevich** [comment]
> if you need admin access to the copr repo

I don't think I need this, but thanks for offering.  I'm afraid I'd likely do more harm than good if I touched it :grin:

**Pull request merged**